pkg/controller/nodemaintenance/nodemaintenance_controller_common_test.go

```diff
@@ -0,0 +1,158 @@
+package nodemaintenance
+
+import (
+
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sdiscovery "k8s.io/client-go/discovery"
+	k8sfakeclient "k8s.io/client-go/kubernetes/fake"
+	kubevirtv1alpha1 "kubevirt.io/node-maintenance-operator/pkg/apis/kubevirt/v1alpha1"
+)
+
+
+func getCommonTestObjs() (*kubevirtv1alpha1.NodeMaintenance, []runtime.Object) {
+		nm := &kubevirtv1alpha1.NodeMaintenance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-maintanance",
+			},
+			Spec: kubevirtv1alpha1.NodeMaintenanceSpec{
+				NodeName: "node01",
+				Reason:   "test reason",
+			},
+		}
+
+		return nm, []runtime.Object{
+			&corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node01",
+//					Finalizers: []string{
+//							kubevirtv1alpha1.NodeMaintenanceFinalizer,
+//						},
+				},
+				Spec: corev1.NodeSpec{
+					Taints: []corev1.Taint{{
+						Key:    "test",
+						Effect: corev1.TaintEffectPreferNoSchedule},
+					},
+				},
+			},
+			&corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node02",
+				},
+			},
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-pod-1",
+				},
+				Spec: corev1.PodSpec{
+					NodeName: "node01",
+				},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:   corev1.PodReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+			},
```

Comment:
FYI, this is another case of defining Go objects inline.
Its quite normal to do.